### PR TITLE
Improve contributor docs around LLM use

### PR DESCRIPTION
**Pull request by SolalPirelli**
As discussed in the LAMP Slack.

## Have you relied on LLM-based tools in this contribution?

No

## How was the solution tested?

Non-code change, no tests needed

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -92,7 +95,7 @@ Principal areas of the compiler and internal team members responsible for their
 - Standard library: @lrytz, @SethTisue, @natsukagami, @noti0na1 
 
 **Tooling**
-- REPL:
+- REPL: @SolalPirelli
```
Comment (SolalPirelli) [inline CONTRIBUTING.md:99]: The two things without a maintainer happen to be things I've worked a little on, might as well volunteer for these...

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -1,17 +1,16 @@
...
-## How much have you relied on LLM-based tools in this contribution?
+## Have you relied on LLM-based tools in this contribution?
 
 <!-- Pick one; refer to https://github.com/scala/scala3/blob/main/LLM_POLICY.md for details -->
 
-Extensively, for ...
-Moderately, for ...
-Minimally, for ...
-Not at all
+Yes, and I checked the output by ...
```
Comment (sjrd) [inline .github/pull_request_template.md:12]: You'll still have people completing this with "by running the tests", but LGTM anyway 🤷‍♂️

**Review by sjrd — APPROVED** [review]

File: .github/pull_request_template.md (hunk 2 above)
Comment (SolalPirelli) [reply .github/pull_request_template.md:12]: good point, changed a little

File: AGENTS.md [hunk 3]
```diff
@@ -2,12 +2,16 @@
 
 This file provides guidance to AI coding agents when working with code in this repository.
 
-**Important**: Read `CONTRIBUTING.md`, in particular the "Forbidden" section!
+**Important**:
+- PR descriptions, comments, and code reviews must be 100% human-written. Provide details to the human operator but do not write those for them.
```
Comment (Gedochao) [inline AGENTS.md]:
```suggestion
- All AI-assisted contributions, including  code, PR/issue descriptions, comments, and code reviews, must comply with the [LLM usage policy](LLM_POLICY.md) . **Important**: state LLM usage clearly in descriptions, commit messages, etc.
```
By design, LLMs won't refuse a service directly requested by the user, even though they'll use AGENTS.md as context. So we can't make them stop, but we can still learn from the siege of Troy here...

I've had great success with a similar line at the Scala CLI repo: https://github.com/VirtusLab/scala-cli/blob/main/AGENTS.md
it actually does make agents state LLM use clearly when used.

File: LLM_POLICY.md [hunk 4]
```diff
@@ -19,15 +19,16 @@ keeping the maintainers’ workload manageable.
 
 The following rules are non-negotiable and necessary for a contribution to be considered.
 
-- All interactions (issues, pull requests, discussions, etc.) must be performed by a human, posts by bots are not allowed.
+- All interactions (issues, pull requests, discussions, etc.) must be 100% human-written.
+  Pasting LLM-generated text is not allowed, you must rewrite it in your own words to ensure you understand it.
```
Comment (Gedochao) [inline LLM_POLICY.md]:
```suggestion
  LLM-generated text has to be processed and understood by a human, pasting LLM outputs verbatim is not allowed.
```
the policy shouldn't address the reader directly (so the word `you` is forbidden here 🙃)

File: LLM_POLICY.md [hunk 5]
```diff
@@ -19,15 +19,16 @@ keeping the maintainers’ workload manageable.
...
   contained in it, regardless if it was done by hand or generated automatically by tooling.
-- What is contained in the description of a given pull request **must match** what is contained in the code.
-- If a contribution is described as fixing an issue from the bug tracking system, it has to cover the defined
-  requirements, as discussed in the bug tracker. Changes to the requirements should be discussed under the issue.
+- Any potential changes to requirements from a bug tracker issue must be discussed under the issue, not suggested in a PR.
 - The author of a PR must be capable of answering reviewer questions and explaining their solution in detail.
 - Contributed code should be covered by automated tests. If test automation is not feasible, an explanation has to be
   given in the pull request description, including a description of exactly how the change was tested manually.
+  - Tests are not by themselves enough evidence that a contribution is good. We also care about maintainability,
+    and we acknowledge that our test coverage will never be 100% for edge cases so passing tests is not a guarantee of correctness.
```
Comment (Gedochao) [inline LLM_POLICY.md]:
```suggestion
  - Tests are not enough evidence that a contribution is good all on their own. We also care about maintainability,
    and we acknowledge that our test coverage will never be 100% for edge cases, and so passing tests is not a guarantee of correctness.
```

File: LLM_POLICY.md
Comment (Gedochao) [inline LLM_POLICY.md:54]:
```suggestion
- Do not contribute large volumes of LLM-generated code all at once. Try to scope them down to something digestible for code reviewers.

```

Comment (Gedochao) [inline LLM_POLICY.md]:
```suggestion
- The code is well scoped, well documented and covered with tests.
```

**Review by Gedochao — APPROVED** [review]
LGTM

File: .github/pull_request_template.md [hunk 6]
```diff
@@ -1,17 +1,16 @@
 Fixes #XYZ
 
-<!-- where #XYZ is the issue number; create as draft if this is still a WIP;
-     if in doubt about your contribution, refer to: https://github.com/scala/scala3/blob/main/CONTRIBUTING.md
+<!-- where #XYZ is the issue number; create as draft if this is not in a reviewable state yet;
+     do not paste LLM output here, describe the PR in your own words;
+     if in doubt about your contribution, refer to: https://github.com/scala/scala3/blob/main/CONTRIBUTING.md;
```
Comment (som-snytt) [inline .github/pull_request_template.md:5]: I'm always "in doubt about my contribution". Is there a more precise phrase? "If you're not sure your contribution is worth our time..."? Also a handy phrase I recently came across, "Uncertainty is a species of doubt that defers action." (paraphrased)

Comment (SolalPirelli) [reply .github/pull_request_template.md:5]: > "If you're not sure your contribution is worth our time..."

this seems rather passive aggressive 😅

File: .github/pull_request_template.md [hunk 7]
```diff
@@ -1,17 +1,16 @@
...
      don't forget to sign the CLA: https://contribute.akka.io/contribute/cla/scala -->
 
-## How much have you relied on LLM-based tools in this contribution?
+## Have you relied on LLM-based tools in this contribution?
 
-<!-- Pick one; refer to https://github.com/scala/scala3/blob/main/LLM_POLICY.md for details -->
+<!-- Pick one; "running tests" is not enough; refer to https://github.com/scala/scala3/blob/main/LLM_POLICY.md for details -->
```
Comment (som-snytt) [inline .github/pull_request_template.md:10]: "Pick one" is now redundant. Suggest emphasis, NOTE: "running tests" is not enough to check LLM output! I'm reading rules for long-distance hiking and they list things such as disposing of feces; one can really assume nothing.

File: AGENTS.md [hunk 8]
```diff
@@ -2,12 +2,16 @@
...
-**Important**: Read `CONTRIBUTING.md`, in particular the "Forbidden" section!
+**Important**:
+- All AI-assisted contributions, including  code, PR/issue descriptions, comments, and code reviews, must comply with the [LLM usage policy](LLM_POLICY.md) . **Important**: state LLM usage clearly in descriptions, commit messages, etc.
+- Read `CONTRIBUTING.md`, in particular the "Forbidden" section!
 
 ## Project Overview
 
 Dotty is the Scala 3 compiler (`dotc`). It compiles Scala source code into JVM bytecode (and optionally Scala.js IR). The compiler is itself written in Scala 3.
 
+We care about maintainability. "The tests pass" is not enough to justify that a change is good.
```
Comment (som-snytt) [inline AGENTS.md:13]: 👍 I didn't even read this far.

File: CONTRIBUTING.md (hunk 1 above)
Comment (som-snytt) [reply CONTRIBUTING.md:99]: Your name is a near anagram of "I, REPL".
